### PR TITLE
Update for Swift 3 and Xcode 8 b3

### PR DESCRIPTION
**Pull request by tonyd256**

### CODE REVIEW

File: bin/generate [hunk 1]
```diff
@@ -36,7 +36,7 @@ func curryDefinitionGenerator(arguments: Int) -> String {
   let genericTypeDefinition = "<\(commaConcat(genericParameters))>" // "<A, B, C, D>"
 
   let inputParameters = Array(genericParameters[0..<arguments - 1]) // ["A", "B", "C"]
-  let lowerFunctionArguments = inputParameters.map { "`\($0.lowercased())`" } // ["`a`", "`b`", "`c`"]
+  let lowerFunctionArguments = inputParameters.map { "\($0.lowercased())" } // ["`a`", "`b`", "`c`"]
```
Comment (jakecraige) [inline bin/generate]: Should the backticks be removed from the comment?

**Comment by jakecraige** [comment]
> It also seems to have improved compile duration by a few seconds
> although this was based off of counting in my head.

😂

**Comment by jakecraige** [comment]
👍

**Comment by gfontenot** [comment]
sure 👍

**Pull request merged**